TaskLayer/SearchTask/SearchTask.cs

```diff
@@ -90,35 +90,42 @@ protected override MyTaskResults RunSpecific(string OutputFolder, List<DbForTask
                 {
                     SearchParameters.DoQuantification = false;
                 }
-                //if we're doing SILAC, add the silac labels to the residue dictionary
-                else if (SearchParameters.SilacLabels != null)
+                //if we're doing SILAC, assign and add the silac labels to the residue dictionary
+                else if (SearchParameters.SilacLabels != null || SearchParameters.StartTurnoverLabel != null || SearchParameters.EndTurnoverLabel != null)
                 {
-                    //change the silac residues to lower case amino acids (currently null)
-                    List<SilacLabel> updatedLabels = new List<SilacLabel>();
-                    char heavyLabel = 'a';
-                    for (int i = 0; i < SearchParameters.SilacLabels.Count; i++)
+                    char heavyLabel = 'a'; //char to assign
+                    //add the Turnoverlabels to the silacLabels list. They weren't there before just to prevent duplication in the tomls
+                    if (SearchParameters.StartTurnoverLabel != null || SearchParameters.EndTurnoverLabel != null)
                     {
-                        SilacLabel currentLabel = SearchParameters.SilacLabels[i];
-                        //make sure we're not overwriting something. , , and if it's a valid residue (not a motif/delimiter)
-                        while ((Residue.TryGetResidue(heavyLabel, out Residue residue) //Check if the amino acid exists. If it already exists, we don't want to overwrite it
-                            && !residue.ThisChemicalFormula.Formula.Equals(currentLabel.LabelChemicalFormula)) //if it exists but it's already the label (so we're not overwriting anything), then we're fine
-                            || GlobalVariables.InvalidAminoAcids.Contains(heavyLabel)) //If it didn't already exist, but it's invalid, we need to keep going
+                        //original silacLabels object is null, so we need to initialize it
+                        SearchParameters.SilacLabels = new List<SilacLabel>();
+                        if (SearchParameters.StartTurnoverLabel != null)
                         {
-                            heavyLabel++;
+                            var updatedLabel = SilacConversions.UpdateAminoAcidLabel(SearchParameters.StartTurnoverLabel, heavyLabel);
+                            heavyLabel = updatedLabel.nextHeavyLabel;
+                            SearchParameters.StartTurnoverLabel = updatedLabel.updatedLabel;
+                            SearchParameters.SilacLabels.Add(SearchParameters.StartTurnoverLabel);
                         }
-                        SilacLabel updatedLabel = SilacConversions.AssignValidHeavyCharacter(currentLabel, heavyLabel);
-                        heavyLabel++;
-                        if (currentLabel.AdditionalLabels != null)
+                        if (SearchParameters.EndTurnoverLabel != null)
```

Comment:
this can just be else since you have SearchParameters.StartTurnoverLabel != null || SearchParameters.EndTurnoverLabel != null to get inside.

Reply:
if both start and end have values, then using else would skip this code.